logic_tree_tools.py

```diff
@@ -0,0 +1,920 @@
+        The initial logic tree pair to modify.
+
+    tectonic_region_type_set : list[str]
+        A list of the selected tectonic region types.
+        Valid tectonic region types are:
+            "Active Shallow Crust",
+            "Subduction Interface",
+            "Subduction Intraslab".
+
+    which_interface : str, default = "HIK_and_PUY"
+        Which subduction interfaces to include.
+        Valid options are:
+           "HIK_and_PUY" which includes HIK_and_PUY the Hikurangi–Kermadec (only_HIK) and Puysegur (only_PUY) subduction zones
+           "only_HIK" which includes only the Hikurangi–Kermadec (only_HIK) subduction zone
+           "only_PUY" which includes only the Puysegur (only_PUY) subduction zone.
+
+    Returns
+    -------
+    modified_logic_tree_pairs : list[CustomLogicTreePair]
+        The modified logic tree pairs that only include branch sets corresponding
+        to the selected tectonic region types.
+    """
+
+    trt_short_lookup_dict = {
+        "Active Shallow Crust": "CRU",
+        "Subduction Interface": "INTER",
+        "Subduction Intraslab": "SLAB",
+    }
+
+    modified_logic_tree_pairs = []
+
+    source_logic_tree = copy.deepcopy(initial_logic_tree_pair.source_logic_tree)
+    ground_motion_logic_tree = copy.deepcopy(
+        initial_logic_tree_pair.ground_motion_logic_tree
+    )
+
+    modified_logic_tree_pair = copy.deepcopy(initial_logic_tree_pair)
+
+    short_tectonic_region_types_for_ground_motion_logic_tree_note = [
+        trt_short_lookup_dict[trt] for trt in tectonic_region_type_set
+    ]
+    short_tectonic_region_types_for_source_logic_tree_note = copy.deepcopy(
+        short_tectonic_region_types_for_ground_motion_logic_tree_note
+    )
+
+    for short_trt_index, short_trt in enumerate(
+        short_tectonic_region_types_for_source_logic_tree_note
+    ):
+        if short_trt == "INTER":
+            short_tectonic_region_types_for_source_logic_tree_note[short_trt_index] = (
+                f"INTER_{which_interface}"
+            )
+
+    modified_source_logic_tree = select_branch_sets_given_tectonic_region_type(
+        source_logic_tree, tectonic_region_type_set, which_interface
+    )
+    modified_ground_motion_logic_tree = select_branch_sets_given_tectonic_region_type(
+        ground_motion_logic_tree, tectonic_region_type_set
+    )
+
+    modified_logic_tree_pair.source_logic_tree_note += f"tectonic_region_type_set:[{' '.join(short_tectonic_region_types_for_source_logic_tree_note)}] > "
+
+    modified_logic_tree_pair.ground_motion_logic_tree_note += f"tectonic_region_type_set:[{' '.join(short_tectonic_region_types_for_ground_motion_logic_tree_note)}] > "
+
+    modified_logic_tree_pair.source_logic_tree = copy.deepcopy(
+        modified_source_logic_tree
+    )
+    modified_logic_tree_pair.ground_motion_logic_tree = copy.deepcopy(
+        modified_ground_motion_logic_tree
+    )
+
+    modified_logic_tree_pairs.append(modified_logic_tree_pair)
```

Comment:
Am I missing something, or will `modified_logic_tree_pairs` only ever contain one value?

Reply:
I think you're right. Good spotting. I think I must have changed my plans for this function halfway through and forgot to undo the list appending part